openshift/console
# ODC-7179 : Allow users to see which pods are receiving traffic

JIRA story:
https://issues.redhat.com/browse/ODC-7179

Description:
Added traffic details for pods in below places in the console UI,
1) In the Resources tab of the topology side panel when showing pods. Along with icon tooltip is also added to indicate whether pod is receiving traffic or not.
2) In the Pods tab of related resources ( Service, Deployment, Deployment Config)
3) In the Pods list view in the context of a namespace, add a traffic column (talk to UX/Sam about how we can add traffic info in the list)
4) In Pods list view / All Namespaces, do not add the traffic column
5) If a Pod is receiving traffic, show that on the Pod details page
 

Demo:


https://user-images.githubusercontent.com/102503482/202167364-98b4c8e9-d5ed-43f4-b599-55e04c3933f7.mov


How to test:
Create the deployment or DC or service and check the traffic details for pods in below places in console, 
1) In the Resources tab of the topology side panel when showing pods. Along with icon tooltip is also added to indicate whether pod is receiving traffic or not.
2) In the Pods tab of related resources ( Service, Deployment, Deployment Config)
3) In the Pods list view in the context of a namespace, add a traffic column (talk to UX/Sam about how we can add traffic info in the list)
4) In Pods list view / All Namespaces, do not add the traffic column
5) If a Pod is receiving traffic, show that on the Pod details page
